### PR TITLE
53899 - bundle info validations

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
Code changes and Unit tests
Functional Test

### CODE REVIEW

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs [hunk 1]
```diff
@@ -388,5 +387,56 @@ public static void CheckProductFileContentLargeFile(string inputFile)
...
+                string productName = item.ProductName;
+                string editionNumber = item.EditionNumber.ToString();
+                //Get Countrycode
+                string countryCode = productName.Substring(0, 2);
+
+                //Get folder path
+                string editionFolderPath = Path.Combine(Path.GetDirectoryName(inputFile), countryCode, productName);
+
+                ////foreach (var updateNumber in item.UpdateNumbers)
+                ////{
```
[the remark this answers is not in the document]
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs]: Updated.Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs [hunk 2]
```diff
@@ -366,16 +366,15 @@ public static async Task<List<string>> CreateExchangeSetFileForLargeMedia(HttpRe
 
             for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
             {
-                var FolderName = $"M0{mediaNumber}X02";
-                var downloadFileUrl = $"{Config.FssConfig.BaseUrl}/batch/{batchId}/files/{FolderName}.zip";
+                    var FolderName = $"M0{mediaNumber}X02";
```
[the remark this answers is not in the document]
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs]: Updated.Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs [hunk 3]
```diff
@@ -348,6 +359,14 @@ private async Task<LargeExchangeSetDataResponse> SearchAndDownloadEncFilesFromFs
             LargeExchangeSetDataResponse largeExchangeSetDataResponse = new LargeExchangeSetDataResponse();
             var response = await DownloadSalesCatalogueResponse(message);
 
+            Task<ValidationResult> result = productDataValidator.Validate(response.Products);
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs]: Resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/InvalidBundleInfoScenariosForLargeMediaExchangeSet.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.FunctionalTests
+{
+    public class InvalidBundleInfoScenariosForLargeMediaExchangeSet
+    {
+        private string EssJwtToken { get; set; }
+        private string FssJwtToken { get; set; }
+        private ExchangeSetApiClient ExchangeSetApiClient { get; set; }
+        private FssApiClient FssApiClient { get; set; }
```
Comment (martyn-fewtrell) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/InvalidBundleInfoScenariosForLargeMediaExchangeSet.cs:16]: Is the FSSApiClient used?

If not can it be removed?

**Review by martyn-fewtrell — COMMENTED** [review]
Just one comment which should be easy to resolve.

I like the approach of moving the validation into a validation object!

It would be good if there are other opportunities to take the same approach as it makes the code very readable!

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/InvalidBundleInfoScenariosForLargeMediaExchangeSet.cs (hunk 4 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/InvalidBundleInfoScenariosForLargeMediaExchangeSet.cs]: Hi Martyn,

The unused code has been removed.

Regards,
Mayuresh Gawde

**Review by martyn-fewtrell — APPROVED** [review]
Thanks for the update.

Looks good to me.

**Pull request merged**